### PR TITLE
Create global loading component

### PR DESCRIPTION
**Pull request by sabicd**

### CODE REVIEW

File: src/components/widgets/app-progress/app-progress.component.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import DialogTitle from '@material-ui/core/DialogTitle';
+
+import Dialog from '@material-ui/core/Dialog';
+import CircularProgress from '@material-ui/core/CircularProgress';
+
+const AppProfile = ({ isAppWaiting }) => (
+  <Dialog
+    aria-labelledby="simple-dialog-title"
+    open={isAppWaiting}
+  >
```
Comment (knightburton) [inline src/components/widgets/app-progress/app-progress.component.js:16]: The dialog should held the props for block the background click and the escape button press.

File: src/store/app/index.js [hunk 2]
```diff
@@ -52,7 +52,7 @@ export const removeAlert = createAction(
  */
 
 export const getIsMobileDrawerOpened = state => state.app.isMobileDrawerOpened;
-export const getAppWaitingCounter = state => state.app.appWaiting;
+export const getAppWaitingCounter = state => state?.app?.appWaiting || 0;
```
Comment (knightburton) [inline src/store/app/index.js]: The base selectors should not be like this. The base selcotr should return the exact value that in the store without any other functionality, should be pure. If you need some logic then create a new selector for it. Also, if you check the counter logic in the reducer, the counter always has a value, and that can be  less than 0. So return a default one is not a good pattern.

File: src/store/profile/index.js [hunk 3]
```diff
@@ -66,7 +66,7 @@ export const reducer = handleActions(
 function* fetchUserDataSaga() {
   yield put(setAppWaiting(true));
   try {
-    yield delay(1000);
+    yield delay(2000);
```
Comment (knightburton) [inline src/store/profile/index.js]: I suggest to use the delay like this.
`yield call(delay, 2000);`

**Review by knightburton — CHANGES_REQUESTED** [review]

File: src/store/app/index.js (hunk 2 above)
Comment (sabicd) [reply src/store/app/index.js]: I see, I left it as I was trying to solve an issue related to the progress component state. Restored the selector to the previous state

**Review by knightburton — APPROVED** [review]

**Pull request merged**